### PR TITLE
Add core superfluid protocol events to daily spellbook

### PR DESCRIPTION
**Pull request by leopk-code**
### Update!
Please build spells in the proper [subproject](../dbt_subprojects/) directory. For more information, please see the main [readme](../README.md), which also links to a GH discussion with the option to ask questions.

### Contribution type
Please check the type of contribution this pull request is for:

- [x] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *dbt_subprojects/daily_spellbook/models/superfluid/**
- **Description:** Seven low-level events that together reflect most of the capabilities of superfluid protocol. 
- **Who are the new spell(s) for?** For Superfluid and community maintained dashboards
- **How will the new spell(s) be used downstream?** Directly in dashboards. Currently no downstream spellbook tables planned
- **Implementation details:** Incremental tables based on the logs table for each network
- **Test instructions:** na
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

File: dbt_subprojects/daily_spellbook/models/superfluid/arbitrum/superfluid_arbitrum_flow_distribution_updated_events.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{{ config(
+    schema = 'superfluid_arbitrum',
+    alias = 'flow_distribution_updated_events',
+    materialized = 'incremental',
```
Comment (Hosuke) [inline dbt_subprojects/daily_spellbook/models/superfluid/arbitrum/superfluid_arbitrum_flow_distribution_updated_events.sql:4]: Hi @leopk-code,
for `incremental` spells, 
we need to define the `unique_key` in config blocks.

File: dbt_subprojects/daily_spellbook/models/superfluid/arbitrum/superfluid_arbitrum_schema.yml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  - name: superfluid_arbitrum_flow_updated_events
+    meta:
+      blockchain: arbitrum
+      sector: defi
+      project: superfluid
+      contributors: superfluid_hq
+    config:
+      tags: ['arbitrum', 'DeFi', 'superfluid']
+    description: >
+      A table containing all flow updated events on Arbitrum.
```
Comment (Hosuke) [inline dbt_subprojects/daily_spellbook/models/superfluid/arbitrum/superfluid_arbitrum_schema.yml:13]: Also we need to test apply 
```
    tests:
      - dbt_utils.unique_combination_of_columns:
          combination_of_columns:
```
test to `unique_key` in model schema.

**Review by Hosuke — CHANGES_REQUESTED** [review]
Thank you for your contribution @leopk-code!
We may add `superfluid_ethereum` roll-up spells to aggregate chain-level spells, then add `post_hook` in roll-up spells' config block.

File: dbt_subprojects/daily_spellbook/models/superfluid/arbitrum/superfluid_arbitrum_flow_distribution_updated_events.sql (hunk 1 above)
Comment (Hosuke) [reply dbt_subprojects/daily_spellbook/models/superfluid/arbitrum/superfluid_arbitrum_flow_distribution_updated_events.sql:4]: dbt docs:
https://docs.getdbt.com/docs/build/incremental-models#defining-a-unique-key-optional

**Comment by cincauhangus** [comment]
Hi there, just FYI @leopk-code has left superfluid and I am taking over his role. 

@leopk-code would it be better if I forked your repo to add the changes (and you'll need to merge it in later) or shall I start a new PR from scratch?

**Comment by leopk-code** [comment]
@cincauhangus might be easier to start a new pull request based on my fork!

**Pull request closed without merging**